chef/automate
# System user for deployment

### :nut_and_bolt: Description: What code changed, and why?
Added implementation to verify system user and group availability on the machine and their correct mapping in order to ensure successful deployment.



### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-2202

### :+1: Definition of Done

### :athletic_shoe: How to Build and Test the Change

1. git checkout this branch
2. `rebuild components/automate-cli/ ` and upload it your origin
3. Install automate-cli on your instance
4. Run `chef-automate verify serve -d`
5. Hit `curl -s "http://localhost:7799/api/v1/checks/system-user" | jq`

Tested on:
Ubuntu
Amazon linux
Suse linux
Centos
Redhat

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

**If User is already present:**
![image](https://github.com/chef/automate/assets/101619541/52768bcd-0ffb-4374-89f0-016e83eac7f2)

![image](https://github.com/chef/automate/assets/101619541/f50577ec-2d70-4f36-ae31-4a3b972a06bc)

**If user not present**
![image](https://github.com/chef/automate/assets/101619541/4e303cd7-af4c-400a-a77b-df0a912d14c8)

![image](https://github.com/chef/automate/assets/101619541/2d0c7fad-d6e4-4ce9-a40a-0d2234de0bd9)

**If primary group is not hab:**
![image](https://github.com/chef/automate/assets/101619541/6c9b2845-2f25-4bba-b268-d7387f5c2046)


![image](https://github.com/chef/automate/assets/101619541/b603e590-aab9-46f0-b175-86db7e649602)

**If user creation failed**
![image](https://github.com/chef/automate/assets/101619541/2e76e192-9fd4-4fec-9479-c048dc2d2c0c)

![image](https://github.com/chef/automate/assets/101619541/ead8438e-f14b-49d8-88a1-78e3395375ac)


Demo video:
https://progresssoftware-my.sharepoint.com/:v:/g/personal/sboorla_progress_com/EWKy59ctPX5Hgsl9hu2lFd4BlpUTHAJ4qBbcJGZmBcszqQ?e=6680sL
